src/app/components/editors/gameobject/gameobject-template/gameobject-template.integration.spec.ts

```diff
@@ -0,0 +1,158 @@
+  '`Data21`, `Data22`, `Data23`, `AIName`, `ScriptName`, `VerifiedBuild`) VALUES\n' +
+  '(' + id + ', 0, 0, \'\', \'\', \'\', \'\', 1, 0, 0, 0, 0, 0, 0, 0, ' +
+  '0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, \'\', \'\', 0);';
+
+  const originalEntity = new GameobjectTemplate();
+  originalEntity.entry = id;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        GameobjectTemplateModule,
+        RouterTestingModule,
+      ],
+    })
+      .compileComponents();
+  }));
+
+  function setup(creatingNew: boolean) {
+    handlerService = TestBed.get(GameobjectHandlerService);
+    handlerService['_selected'] = `${id}`;
+    handlerService.isNew = creatingNew;
+
+    queryService = TestBed.get(QueryService);
+    querySpy = spyOn(queryService, 'query').and.returnValue(of());
+
+    spyOn(queryService, 'selectAll').and.returnValue(of(
+      { results: creatingNew ? [] : [originalEntity] }
+    ));
+
+    fixture = TestBed.createComponent(GameobjectTemplateComponent);
+    component = fixture.componentInstance;
+    page = new GameobjectTemplatePage(fixture);
+    fixture.autoDetectChanges(true);
+    fixture.detectChanges();
+  }
+
+  describe('Creating new', () => {
+    beforeEach(() => setup(true));
+
+    it('should correctly initialise', () => {
+      page.expectQuerySwitchToBeHidden();
+      page.expectFullQueryToBeShown();
+      page.expectFullQueryToContain(expectedFullCreateQuery);
+    });
+
+    it('changing a property and executing the query should correctly work', () => {
+      const expectedQuery = 'DELETE FROM `gameobject_template` WHERE (`entry` = ' + id + ');\n' +
+      'INSERT INTO `gameobject_template` (`entry`, `type`, `displayId`, `name`, `IconName`, `castBarCaption`, ' +
+      '`unk1`, `size`, `Data0`, `Data1`, `Data2`, `Data3`, `Data4`, `Data5`, `Data6`, `Data7`, `Data8`, `Data9`, ' +
+      '`Data10`, `Data11`, `Data12`, `Data13`, `Data14`, `Data15`, `Data16`, `Data17`, `Data18`, `Data19`, `Data20`, ' +
+      '`Data21`, `Data22`, `Data23`, `AIName`, `ScriptName`, `VerifiedBuild`) VALUES\n' +
+      '(' + id + ', 0, 0, \'Helias\', \'\', \'\', \'\', 1, 0, 0, 0, 0, 0, 0, 0, ' +
+      '0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, \'\', \'\', 0);';
+
+      querySpy.calls.reset();
+
+      page.setInputValueById('name', 'Helias');
+      page.clickExecuteQuery();
+
+      page.expectFullQueryToContain(expectedQuery);
+      expect(querySpy).toHaveBeenCalledTimes(1);
+      expect(querySpy.calls.mostRecent().args[0]).toContain(expectedQuery);
+    });
+  });
+
+  describe('Editing existing', () => {
+    beforeEach(() => setup(false));
+
+    it('should correctly initialise', () => {
+      page.expectDiffQueryToBeShown();
+      page.expectDiffQueryToBeEmpty();
+      page.expectFullQueryToContain(expectedFullCreateQuery);
+    });
+
+    it('changing a property and executing the query should correctly work', () => {
```

Comment:
please change this test (and its logic) to:

```'changing all properties and executing the query should correctly work'```

see: https://github.com/azerothcore/Keira3/pull/46